src/main/java/core/basesyntax/servise/operation/PurchaseOperation.java

```diff
@@ -0,0 +1,10 @@
+package core.basesyntax.servise.operation;
+
+import java.util.Map;
+
+public class PurchaseOperation implements OperationHandler {
+    @Override
+    public void apply(String fruit, int quantity, Map<String, Integer> storage) {
+        storage.put(fruit, storage.getOrDefault(fruit, 0) - quantity);
+    }
```

Comment:
Handling Purchase operation requires checking the result balance before saving it in the Storage to ensure it is positive. If the balance is negative after the purchase, you should throw a RuntimeException to prevent invalid transactions.